### ISSUE TITLE
Search for station, select it, and get station

### ISSUE THREAD

**Issue by mspencer92** [mcrute/pydora#63]
How would I go about getting a station after searching for it?
I currently have the code:
`search = pandora_client.search("Today's Hits", include_near_matches=True, include_genre_stations=True)`
`station = search[0]`
`station.create_station()`

After creating the station; how do I select it to get a Station and get its tracks using get_playlist()?

**Comment by mspencer92** [comment]
Modifying `models/search.py` the `GenreStationSearchResultItem` classes function `create_station` to return `self._api_client.create_station(search_token=self.token)` does the trick, should I make a pull request to make the default?

**Comment by mspencer92** [comment]
With that I can just do, `station = station.create_station()` with the above code and it will return the created station.

**Comment by mcrute** [comment]
That looks like a mistake in the API design. I'd be happy to accept a pull request to fix it. Would you also mind updating the other `create_station` methods to return as well while you're in there? Also would you please update the [model tests](https://github.com/mcrute/pydora/blob/master/tests/test_pandora/test_models.py) to expect that return value?

**Comment by mspencer92** [comment]
I can do that; I have also found another issue while creating a station from ArtistSearchResultItem, receiving an `pandora.errors.InternalServerError: Internal Server Error`.

`search = client.search("Britney Spears")
for station in search.artists:
    station = station.create_station()
    for song in station.get_playlist():
        print(song)`

Error is coming from the station.create_station.

**Comment by mspencer92** [comment]
Fixed the above error by changing this code inside `pandora/client.py`:
```        elif artist_token:
            kwargs = {"trackToken": artist_token, "musicType": "artist"}
```
I changed `trackToken` to `musicToken` and the error went away. If this is not the correct fix please let me know.

**Comment by mspencer92** [comment]
Same issue with SongSearchResultItem, but also fixed by changing trackToken to musicToken 😄

**Comment by mcrute** [comment]
Search is one of the few parts of the API that I haven't put almost any end-to-end testing into personally, though it's been perpetually on my list. I think other downstream projects may have but I haven't seen a lot of patches to it. Any improvements you're willing to contribute there would be very much appreciated.

**Comment by mspencer92** [comment]
> Search is one of the few parts of the API that I haven't put almost any end-to-end testing into personally, though it's been perpetually on my list. I think other downstream projects may have but I haven't seen a lot of patches to it. Any improvements you're willing to contribute there would be very much appreciated.

I will certainly make a pull request with my found fixes and try my best to do the model tests that I don't have any experience with. If I find any other problems while implementing pydora into my current project I will let you know. Thank you for the help.

**Comment by mspencer92** [comment]
I am currently programming a discord bot that plays pandora stations in the voice channel. Not sure of the legality of that but I am sure it is fine.

**Comment by mcrute** [comment]
> I will certainly make a pull request with my found fixes and try my best to do the model tests that I don't have any experience with. If I find any other problems while implementing pydora into my current project I will let you know. Thank you for the help.

Thanks! If you get stuck on the testing just post a PR and I'm happy to help get the test fixed up.

**Comment by mcrute** [comment]
All I can say about specific use-cases is that you should check the Pandora terms of service and make sure you follow them. This library is here for people to make neat stuff, not violate their TOS.

**Comment by mspencer92** [comment]
> All I can say about specific use-cases is that you should check the Pandora terms of service and make sure you follow them. This library is here for people to make neat stuff, not violate their TOS.

Gotcha, I will do that.

**Comment by mcrute** [comment]
Pushed the fixes for this issue as 2.1.0
